Adds Eigen pitfalls link (#143)
NUbots/NUbook: src/book/02-system/01-foundations/05-mathematics.mdx

## Linear Algebra

> We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase. Be aware that Eigen has some unique pitfalls and these are listed on the [Eigen pitfalls page](https://eigen.tuxfamily.org/dox/TopicPitfalls.html).

Author: ```suggestion
We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase. Be aware that Eigen has some common pitfalls that are listed on the [Eigen pitfalls page](https://eigen.tuxfamily.org/dox/TopicPitfalls.html).
```
